api/views.py

```diff
@@ -67,10 +67,13 @@ def create(self, request, **kwargs):
                 answer.status = Answer.CORRECT
                 puzzle.answer = answer.text
                 if puzzle.chat_room:
-                    puzzle.chat_room.archive_channels()
-                    msg = f"{puzzle.name} has been solved with {answer.text}!"
-                    puzzle.chat_room.get_service().announce(msg)
-                    puzzle.chat_room.send_message(msg)
+                    try:
+                        puzzle.chat_room.archive_channels()
+                        msg = f"{puzzle.name} has been solved with {answer.text}!"
+                        puzzle.chat_room.get_service().announce(msg)
+                        puzzle.chat_room.send_message(msg)
+                    except:
+                        logger.warn("Chat operations failed.")
```

Comment:
Can we add a test to ensure that if any of these methods raise an error, we still create objects in the db?